hawkbit-security-core/src/main/java/org/eclipse/hawkbit/security/SystemSecurityContext.java

```diff
@@ -111,6 +114,45 @@ public SystemSecurityContext(final TenantAware tenantAware) {
         }
     }
 
+    /**
+     * Runs a given {@link Callable} within a system security context, which has
+     * the provided {@link GrantedAuthority}s to successfully run the
+     * {@link Callable}.
+     * 
+     * The security context will be switched to the a new {@link SecurityContext} and back after
+     * the callable is called.
+     * 
+     * @param callable
+     *            to call within the security context
+     * @param authorities
+     *            with which the code block needs to be executed
+     * @param tenant
+     *            to act as system code
+     * @return
+     */
+    public <T> T runWithAuthority(final Callable<T> callable, final Collection<? extends GrantedAuthority> authorities,
```

Comment:
can you please explain me shortly why this is needed?

Reply:
**Background:** It seems to be a practice in the hawkbit to protect the ControllerManagement methods with IS_CONTROLLER role. And other services accessing it must access it under IS_CONTROLLER context. Refer to `AmqpMessageHandlerService::onMessage`

**Why is this needed**: We are accessing controllerManagement in multiple places. Replicating the code written in `AmqpMessageHandlerService::onMessage` was not a good idea. So created a re-usable `runWithAuthority` under SystemSecurityContext.java.